src/execution_impl.rs

```diff
@@ -13,88 +16,119 @@ use wasmer::{
     sub_remaining_point(env, settings::METERING.call_price()).unwrap();
-    let value = exec(
+    let module = &get_module(address).unwrap();
+    exec(
         get_remaining_points(env),
         None,
         module,
-        fnc,
+        function,
         &[],
         &env.interface,
     )
-    .unwrap();
+    .unwrap() // TODO: Should return a Result<Response>
+}
+
+/// Raw call that have the right type signature to be able to be call a module
+/// directly form AssemblyScript:
+///
+#[doc = include_str!("../wasm/README.md")]
+pub fn assembly_script_call_module(env: &Env, address: i32, function: i32) -> i32 {
+    let memory = env.wasm_env.memory.get_ref().expect("initialized memory");
+    // TODO: replace all unwrap() by expect()
+    let addr_ptr = StringPtr::new(address as u32);
+    let func_ptr = StringPtr::new(function as u32);
+    let address = &addr_ptr.read(memory).unwrap();
+    let function = &func_ptr.read(memory).unwrap();
+    let value = call_module(env, address, function);
     let ret = StringPtr::alloc(&value.ret, &env.wasm_env).unwrap();
     ret.offset() as i32
 }
 
-/// Create an instance of VM from a module with a
-/// given intefrace, an operation number limit and a webassembly module
+/// `Print` ABI called by the webassembly VM
 ///
-fn create_instance(limit: u64, module: &[u8], interface: &Interface) -> Result<Instance> {
+/// An utility print function to write on stdout directly from AssemblyScript:
+pub fn assembly_script_print(env: &Env, arg: i32) {
+    let str_ptr = StringPtr::new(arg as u32);
+    // TODO: may be in wasmer-as create a pointer that keep memory?
+    println!(
+        "{}",
+        str_ptr
+            .read(env.wasm_env.memory.get_ref().expect("initialized memory"))
+            .unwrap()
+    );
+}
+
+/// Create an instance of VM from a module with a given interface, an operation
+/// number limit and a webassembly module
+pub fn create_instance(limit: u64, module: &[u8], interface: &Interface) -> Result<Instance> {
     let metering = Arc::new(Metering::new(limit, |_: &Operator| -> u64 { 1 }));
     let mut compiler_config = Cranelift::default();
     compiler_config.push_middleware(metering);
     let store = Store::new(&Universal::new(compiler_config).engine());
     let resolver: ImportObject = imports! {
         "env" => {
-            "abort" =>  Function::new_native_with_env(&store, Env::new(interface), abort)
+            "abort" =>  Function::new_native_with_env(&store, Env::new(interface), assembly_script_abort),
         },
-        "index" => {
-            "print" => Function::new_native_with_env(&store, Env::new(interface), print),
-            "call" => Function::new_native_with_env(&store, Env::new(interface), call),
+        "massa" => {
+            "call_module" =>  Function::new_native_with_env(&store, Env::new(interface), assembly_script_call_module),
+            "print" => Function::new_native_with_env(&store, Env::new(interface), assembly_script_print),
```

Comment:
Does that mean that from the inside of a SC, when I use print, it will call assembly_script_print ?

Reply:
we'll provide a nice library :stars:  so the user will just write

```ts
import { print } from "massa.ts"
print("yeah heeee")
```

Reply:
yay :tada: